code/modules/mob/living/simple_animal/hostile/terror_spiders/actions.dm

```diff
@@ -0,0 +1,127 @@
+
+
+/datum/action/innate/terrorspider/web
+	name = "Web"
+	icon_icon = 'icons/effects/effects.dmi'
+	button_icon_state = "stickyweb1"
+
+/datum/action/innate/terrorspider/web/Activate()
+	var/mob/living/simple_animal/hostile/poison/terror_spider/user = owner
+	user.Web()
+
+/datum/action/innate/terrorspider/wrap
+	name = "Wrap"
+	icon_icon = 'icons/effects/effects.dmi'
+	button_icon_state = "cocoon_large1"
+
+/datum/action/innate/terrorspider/wrap/Activate()
+	var/mob/living/simple_animal/hostile/poison/terror_spider/user = owner
+	user.FindWrapTarget()
+	user.DoWrap()
+
+
+// ---------- WEB
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/Web()
+	visible_message("<span class='notice'>\the [src] begins to secrete a sticky substance.</span>")
+	if(do_after(src, 40, target = loc))
+		new /obj/effect/spider/terrorweb(loc)
+
+/obj/effect/spider/terrorweb
+	name = "terror web"
+	desc = "it's stringy and sticky"
+	icon = 'icons/effects/effects.dmi'
+	anchored = 1 // prevents people dragging it
+	density = 0 // prevents it blocking all movement
+	health = 20 // two welders, or one laser shot (15 for the normal spider webs)
+	icon_state = "stickyweb1"
+
+
+/obj/effect/spider/terrorweb/New()
+	if(prob(50))
+		icon_state = "stickyweb2"
+
+
+/obj/effect/spider/terrorweb/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
+	if(air_group || (height==0)) return 1
```

Comment:
`height` and `air_group` are redundant with LINDA; Please remove them from the args (leaving `/obj/effect/spider/terrorweb/CanPass(atom/movable/mover, turf/target)` and remove this conditional.